owncloudApp/src/main/java/com/owncloud/android/presentation/transfers/TransferListComposeFragment.kt

```diff
@@ -0,0 +1,532 @@
+                        Text(
+                            text = stringResource(id = R.string.action_upload_retry).uppercase(),
+                            color = colorResource(id = R.color.white),
+                        )
+                    }
+                }
+
+            }
+        }
+        if (status == TransferStatus.TRANSFER_SUCCEEDED || status == TransferStatus.TRANSFER_FAILED){
+            Divider(
+                modifier = Modifier
+                    .fillMaxWidth(),
+                color = colorResource(id = R.color.filelist_icon_background),
+                thickness = 1.dp
+            )
+        }
+
+    }
+
+    private fun headerTitleStringRes(status: TransferStatus): Int {
+        return when (status) {
+            TransferStatus.TRANSFER_IN_PROGRESS -> R.string.uploads_view_group_current_uploads
+            TransferStatus.TRANSFER_FAILED -> R.string.uploads_view_group_failed_uploads
+            TransferStatus.TRANSFER_SUCCEEDED -> R.string.uploads_view_group_finished_uploads
+            TransferStatus.TRANSFER_QUEUED -> R.string.uploads_view_group_queued_uploads
+        }
+    }
+
+    private fun checkAccount(transfer: OCTransfer): String {
+        return try {
+            val account = AccountUtils.getOwnCloudAccountByName(requireContext(), transfer.accountName)
+            val oca = OwnCloudAccount(account, requireContext())
+            val accountName = oca.displayName + " @ " +
+                    DisplayUtils.convertIdn(account.name.substring(account.name.lastIndexOf("@") + 1), false)
+            accountName
+        } catch (e: Exception) {
+            Timber.w("Couldn't get display name for account, using old style")
+            transfer.accountName
+        }
+    }
+
+    private fun retryUpload(transfer: OCTransfer) {
+        if (transfer.lastResult == TransferResult.CREDENTIAL_ERROR) {
+            val parentActivity = requireActivity() as FileActivity
+            val account = AccountUtils.getOwnCloudAccountByName(requireContext(), transfer.accountName)
+            parentActivity.fileOperationsHelper.checkCurrentCredentials(account)
+        } else {
+            val file = File(transfer.localPath)
+            if (file.exists()) {
+                transfersViewModel.retryUploadFromSystem(transfer.id!!)
+            } else if (DocumentFile.isDocumentUri(requireContext(), Uri.parse(transfer.localPath))) {
+                transfersViewModel.retryUploadFromContentUri(transfer.id!!)
+            } else {
+                showMessageInSnackbar(getString(R.string.local_file_not_found_toast))
+            }
+        }
+    }
+
+    private fun checkProgress(transfer: OCTransfer, workInfo: List<WorkInfo>): Int {
+        workInfo.forEach { workInfo ->
+            if (workInfo.tags.contains(transfer.id.toString())) {
+                return workInfo.progress.getInt(DownloadFileWorker.WORKER_KEY_PROGRESS, -1)
+            }
+        }
+        return 0
+    }
+}
+
+
+
+
+
+
+
+
```

Comment:
Too many blank lines at the end of the file, 1 blank line is enough 😄

Reply:
Ok! Now, there is only one blank line at the end 👌